base/cvd/cuttlefish/host/commands/cvd/unittests/selector/instance_database_test.cpp

```diff
@@ -476,5 +476,50 @@ TEST_F(CvdInstanceDatabaseJsonTest, DumpLoadDumpCompare) {
   }
 }
 
+TEST_F(CvdInstanceDatabaseTest, UpdateInstances) {
+  if (!SetUpOk()) {
+    GTEST_SKIP() << Error().msg;
+  }
+  auto& db = GetDb();
+
+  cvd::InstanceGroup grp;
+  grp.set_home_directory(Workspace() + "/grp1_home");
+  grp.set_name("grp1");
+  auto ins1 = grp.add_instances();
+  ins1->set_name("ins1");
+  ins1->set_state(cvd::InstanceState::PREPARING);
+  auto ins2 = grp.add_instances();
+  ins2->set_name("ins2");
+  ins2->set_state(cvd::InstanceState::PREPARING);
+
+  auto add_res = db.AddInstanceGroup(grp);
+  ASSERT_TRUE(add_res.ok())
+      << "Failed to add group to db: " << add_res.error().Message();
+
+  auto instance_group = *(std::move(add_res));
+  ASSERT_TRUE(instance_group.ProductOutPath().empty());
+  instance_group.SetProductOutPath("/path/to/product");
+  auto& instance1 = instance_group.Instances()[0];
+  instance1.SetInstanceId(1);
+  instance1.SetState(cvd::InstanceState::STARTING);
+  auto& instance2 = instance_group.Instances()[1];
+  instance2.SetInstanceId(2);
+  instance2.SetState(cvd::InstanceState::STARTING);
+
+  auto update_res = db.UpdateInstanceGroup(instance_group);
+  ASSERT_TRUE(update_res.ok())
+      << "Failed to update database: " << update_res.error().Message();
+
+  auto find_res = db.FindGroup(Query("group_name", "grp1"));
+  ASSERT_TRUE(find_res.ok()) << find_res.error().Message();
+
+  ASSERT_EQ(find_res->HomeDir(), Workspace() + "/grp1_home");
+  ASSERT_EQ(find_res->ProductOutPath(), "/path/to/product");
+  ASSERT_EQ(find_res->Instances()[0].InstanceId(), 1);
+  ASSERT_EQ(find_res->Instances()[1].InstanceId(), 2);
+  ASSERT_EQ(find_res->Instances()[0].State(), cvd::InstanceState::STARTING);
+  ASSERT_EQ(find_res->Instances()[1].State(), cvd::InstanceState::STARTING);
```

Comment:
This final `ASSERT` block could use `EXPECT` which is [non-fatal](https://google.github.io/googletest/primer.html#assertions). The test still fails but the other `EXPECT` calls will still be run and can provide additional information in the test results.

Reply:
Done